### ISSUE TITLE
Test error with latest mongoose version

### ISSUE THREAD

**Issue by lcalvy** [wprl/baucis#324]
Hi,

Doing this sequence fail on test:
```
git clone git@github.com:wprl/baucis.git
npm i mongoose express
npm test
```

```
  Headers
    ✓ sets Last-Modified for single documents
    ✓ sets Last-Modified for the collection
    ✓ sets Etag for single documents
    ✓ sets Etag for the collection
    ✓ sets Allowed
Not Acceptable: The requested content type could not be provided (406).
Error: Can't set headers after they are sent.
    at ServerResponse.setHeader (_http_outgoing.js:371:11)
    at vary (/Users/lcalvy/Documents/Sources/baucis/node_modules/vary/index.js:129:9)
    at ServerResponse.res.vary (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/response.js:930:3)
    at ServerResponse.res.format (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/response.js:630:8)
    at Function.baucis.formatters (/Users/lcalvy/Documents/Sources/baucis/index.js:55:12)
    at /Users/lcalvy/Documents/Sources/baucis/Controller/errors.js:105:12
    at Layer.handle_error (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/layer.js:71:5)
    at trim_prefix (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:315:13)
    at /Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:335:12)
    at next (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:275:10)
    at /Users/lcalvy/Documents/Sources/baucis/Controller/errors.js:92:5
    at Layer.handle_error (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/layer.js:71:5)
    at trim_prefix (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:315:13)
    at /Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:335:12)
    at next (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:275:10)
    at /Users/lcalvy/Documents/Sources/baucis/Controller/errors.js:83:65
    at Layer.handle_error (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/layer.js:71:5)
    at trim_prefix (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:315:13)
    at /Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:335:12)
    at next (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:275:10)
    at /Users/lcalvy/Documents/Sources/baucis/Controller/errors.js:75:68
    at Layer.handle_error (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/layer.js:71:5)
    at trim_prefix (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:315:13)
    at /Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:335:12)
    at next (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:275:10)
    at /Users/lcalvy/Documents/Sources/baucis/Controller/errors.js:49:7
    at Layer.handle_error (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/layer.js:71:5)
    at trim_prefix (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:315:13)
    at /Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:335:12)
    at next (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:275:10)
    at /Users/lcalvy/Documents/Sources/baucis/Controller/errors.js:36:29
    at Layer.handle_error (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/layer.js:71:5)
    at trim_prefix (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:315:13)
    at /Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:335:12)
    at Immediate.next (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:275:10)
    at Immediate.<anonymous> (/Users/lcalvy/Documents/Sources/baucis/node_modules/express/lib/router/index.js:635:15)
    at runCallback (timers.js:674:20)
    at tryOnImmediate (timers.js:645:5)
    at processImmediate [as _immediateCallback] (timers.js:617:5)
    1) sends 406 Not Acceptable when the requested type is not accepted


  69 passing (4s)
  1 failing

  1) Headers sends 406 Not Acceptable when the requested type is not accepted:
     Error: timeout of 2000ms exceeded. Ensure the done() callback is being called in this test.
  

```
Using nodejs v7.10.1

**Comment by pjmolina** [comment]
+1 Same behavior here. 

```
Headers
  Error: Can't set headers after they are sent.	
  1) Headers sets Last-Modified for the collection:

  Error: Can't set headers after they are sent.	
  2) POST plural should handle malformed JSON inside first-level objects but ignore those outside:
```

Hypothesis: a breaking change in how `express` expect the headers to be set?

Reproduced with express: v. 4.16.1 and mongoose v. 4.11.14

**Comment by wprl** [comment]
Looks like this is fixed in 1.6.8
